### PR TITLE
Main tabs structure

### LINKED ISSUE #153: Restructure Buy/Sell component
Separate the Buy/Sell buttons from the Filter Panel, so they can exist independently. This would allow us to develop the Storage pages which do not always have the filter panel and its a better and clearer design.

### PR DESCRIPTION
**Pull request by farinalvaro**
This PR closes #153

Updates `rif-ui` dependency to `0.1.9`

~Also depends on a new version of `rif-ui` that should be released with [this rif-ui PR](https://github.com/rsksmart/rif-ui/pull/128)~

### CODE REVIEW

File: src/components/pages/storage/StorageRoutes.tsx [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
+      label: 'My purchases',
+      to: '/storage/mypurchases',
+      value: '/storage/mypurchases',
+    },
+  ]
+  return (
+    // TODO: define value at root level of every subpage, for example
+    // on storage/sell/done, the value should be storage/sell because
+    // there's no tab with value sell/done and the console will throw an error
+    <TabsPageTemplate title="Storage" value={history.location.pathname} tabs={tabs}>
```
Comment (farinalvaro) [inline src/components/pages/storage/StorageRoutes.tsx:46]: For now I've been thinking of creating a function that given a url returns a valid value. If someone has better solution please suggest 🙂

File: src/components/pages/storage/StorageRoutes.tsx [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+      label: 'My purchases',
+      to: ROUTES.STORAGE.MYPURCHASES.BASE,
+      value: ROUTES.STORAGE.MYPURCHASES.BASE,
+    },
+  ]
 
-  if (storageEnabled) {
-    return (
+  const getTabValueFromLocation = () => {
+    const { location: { pathname } } = history
```
Comment (jurajpiar) [inline src/components/pages/storage/StorageRoutes.tsx:43]: Prefer useLocation over useHistory.

Comment (jurajpiar) [inline src/components/pages/storage/StorageRoutes.tsx:43]: Extract component

File: src/components/pages/storage/buy/StorageOffersPage.tsx [hunk 3]
```diff
@@ -62,18 +51,6 @@ const StorageOffersPage: FC = () => {
...
-  const switchTxType = () => {
-    if (!txType) {
-      setTxType(1)
-      history.replace(ROUTES.STORAGE.SELL.BASE)
-      return
-    }
-    setTxType(0)
-    history.replace(ROUTES.STORAGE.BUY.BASE)
-  }
-
```
Comment (jurajpiar) [inline src/components/pages/storage/buy/StorageOffersPage.tsx:76]: I think these changes are unnecessary for this PR and are going to cause unnecessary merge conflicts with my PR. Could you drop this file from this PR, or is there a good reason for keeping it?

File: src/components/pages/storage/sell/StorageListingPage.tsx [hunk 4]
```diff
@@ -199,33 +179,29 @@ const StorageListingPage = () => {
...
-          && (
-            <div className={classes.progressContainer}>
-              <TransactionInProgressPanel
-                {...{ isPendingConfirm, onProcessingComplete }}
-                text="Listing your offer!"
-                progMsg="The waiting period is required to securely list your offer.
+    <CenteredPageTemplate
+      title="List storage service"
+      subtitle="Fill out the form below to list your service. All information provided is meant to be true and correct."
+    >
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/StorageListingPage.tsx:185]: This one won't be such an issue but still, isn't this (file) outside of the scope of this PR? (no action needed unless you think that for consistency this could be included in a different PR)

File: src/components/templates/CenteredPageTemplate.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import React, { FC } from 'react'
```
Comment (jurajpiar) [inline src/components/templates/CenteredPageTemplate.tsx:1]: same as in src/components/pages/storage/sell/StorageListingPage.tsx

File: src/components/pages/storage/sell/StorageListingPage.tsx (hunk 4 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/StorageListingPage.tsx:185]: I see your point and yes, it could be done in a different PR. But as I needed to wrap the content of the page and center it, I thought it was a good opportunity to create a template for it as we would probably use it in many pages

File: src/components/pages/storage/buy/StorageOffersPage.tsx (hunk 3 above)
Comment (farinalvaro) [reply src/components/pages/storage/buy/StorageOffersPage.tsx:76]: I could drop this file to avoid conflicts, sure, but I think this is inside of the scope of this PR as we now want to navigate using the new tabs, right?

File: src/components/pages/storage/StorageRoutes.tsx (hunk 2 above)
Comment (farinalvaro) [reply src/components/pages/storage/StorageRoutes.tsx:43]: Good point, thanks

File: src/components/pages/storage/sell/StorageListingPage.tsx (hunk 4 above)
Comment (jurajpiar) [reply src/components/pages/storage/sell/StorageListingPage.tsx:185]: of course. makes sense

File: src/components/pages/storage/buy/StorageOffersPage.tsx (hunk 3 above)
Comment (jurajpiar) [reply src/components/pages/storage/buy/StorageOffersPage.tsx:76]: ditto

File: src/components/pages/rns/RnsTabedPages.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            </Switch>
+          </RnsDomainsStoreProvider>
+        </Route>
+        <Route component={NotFound} />
+      </Switch>
+    </TabsTemplate>
+  )
+}
+
+export default RNSTabedPages
```
Comment (jurajpiar) [inline src/components/pages/rns/RnsTabedPages.tsx]: I was thinking more in terms of a reusable component or HOC.
Something like this perhaps?
```

interface Options {
  tabs: StyledNavTabProps[]
  title: string
}

const withTabs = ({ tabs, title }: Options) => (WrappedComponent: React.ElementType) => {
  const location = useLocation()

  const getTabValueFromLocation = () => {
    const { pathname } = location
    const activeTab = tabs.find((tab) => pathname.includes(tab.to))
    return activeTab?.to || ROUTES.LANDING
  }

  return (
    <TabsTemplate
      title={title}
      value={getTabValueFromLocation()}
      tabs={tabs}
    >
      <WrappedComponent />
    </TabsTemplate>
  )
}

export default withTabs
```

Then, in storage for example:
```
return (
      <Switch>
        <Redirect exact from={ROUTES.STORAGE.BASE} to={ROUTES.STORAGE.BUY.BASE} />
        {
          withTabs({ tabs: TABS, title: 'Storage' })(() =>
            <Switch>
              <Route exact path={ROUTES.STORAGE.BUY.BASE} component={StorageOffersPage} />
              <Route exact path={ROUTES.STORAGE.SELL.BASE}>
                <StorageListingStoreProvider>
                  <StorageListingPage />
                </StorageListingStoreProvider>
              </Route>
              <Route exact path={ROUTES.STORAGE.SELL.DONE} component={StorageOfferListed} />
              <Route component={NotFound} />
            </Switch>
          )
        }
      </Switch>
    )
  }
```

File: src/components/pages/storage/StorageTabedPages.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          </StorageListingStoreProvider>
+        </Route>
+        <Route exact path={ROUTES.STORAGE.SELL.DONE} component={StorageOfferListed} />
+        <Route component={NotFound} />
+      </Switch>
+    </TabsTemplate>
+  )
+}
+
+export default StorageTabedPages
```
Comment (jurajpiar) [inline src/components/pages/storage/StorageTabedPages.tsx]: same as for rns

**Review by jurajpiar — APPROVED** [review]
